pulp_rpm/app/serializers.py

```diff
@@ -458,3 +463,47 @@ def validate(self, data):
             new_data.update({'types': final_types})
 
         return new_data
+
+
+class ModulemdSerializer(SingleArtifactContentSerializer):
+    """
+    Modulemd serializer.
+    """
+
+    name = serializers.CharField(
+        help_text=_("Modulemd name."),
+    )
+    stream = serializers.CharField(
+        help_text=_("Stream name."),
+    )
+    version = serializers.CharField(
+        help_text=_("Modulemd version."),
+    )
+    context = serializers.CharField(
+        help_text=_("Modulemd context."),
+    )
+    arch = serializers.CharField(
+        help_text=_("Modulemd architecture."),
+    )
+    artifacts = serializers.CharField(
+        help_text=_("Modulemd artifacts (rpm packages)."),
+        allow_null=True
+    )
+    dependencies = serializers.CharField(
+        help_text=_("Modulemd dependencies."),
+        allow_null=True
+    )
+    _packages = serializers.PrimaryKeyRelatedField(
+        help_text=_("Pulp connected packages."),
```

Comment:
How about `Modulemd artifacts' packages` or `Modulemd artifacts' content`?
_packages also should be read-only, users should not be able to create/update it via API.

Reply:
I agree sounds better.